runtime/gc_stats.c

```diff
@@ -101,13 +102,20 @@ void caml_orphan_alloc_stats(caml_domain_state *domain) {
   caml_plat_unlock(&orphan_lock);
 }
 
-
 /* The "sampled stats" of a domain are a recent copy of its
    domain-local stats, accessed without synchronization and only
    updated ("sampled") during stop-the-world events -- each minor
    collection, major cycle (which potentially includes compaction),
    all of these events could happen during domain termination. */
-static struct gc_stats sampled_gc_stats[Max_domains];
+static struct gc_stats* sampled_gc_stats;
```

Comment:
Out of curiosity, is there a reason for not having this as a field in the domain state variable?

Reply:
There are a few more failure cases to handle if `sampled_gc_stats` were moved to `Caml_state`. In particular, `Caml_state` is created on demand when domains are created. If `gc_stats*` were a field in `Caml_state`, even with the guarantee that stats are only sampled in stop-the-world sections, we need to check for whether `gc_stats` is not NULL (since newly created domains may not participate in stw sections).

I tried to make this change, but the changes for handling failure cases got messy. Perhaps one solution is to have `sampled_gc_stats` in `dom_internal` struct. But that struct is internal to `domain.c`. 

Something to come back to later.

Reply:
ok, sounds like a good enough reason